apis/apiextensions/v1alpha1/composition_types.go

```diff
@@ -302,7 +302,16 @@ type ConnectionDetail struct {
 
 	// FromConnectionSecretKey is the key that will be used to fetch the value
 	// from the given target resource.
-	FromConnectionSecretKey string `json:"fromConnectionSecretKey"`
+	// +optional
+	FromConnectionSecretKey *string `json:"fromConnectionSecretKey,omitempty"`
+
+	// Value that will be propagated to the connection secret of the composition
+	// instance. Typically you should use FromConnectionSecretKey instead, but
+	// an explicit value may be set to inject a fixed, non-sensitive connection
+	// secret values, for example a well-known port. Supercedes
+	// FromConnectionSecretKey when set.
+	// +optional
+	Value []byte `json:"value,omitempty"`
```

Comment:
Why `[]byte`?

Reply:
I figured it was more consistent with the kind of data it could be used for, which will be written to a secret and live alongside things read from a secret. It is a pretty gross experience to have to write what will very often be strings as base64 encoded data though; I could be convinced to switch it to `*string` to make it friendlier.

Reply:
Yeah, I think `*string` would be much more user friendly. AFAIK, the reason secrets have base64-encoded value is about the ability to store binary data. While in ConfigMaps, they introduced `binaryData` field for base64-encoded data alongside normal string data. I doubt it, but if in the future we need to store binary data, it could be another field.

Reply:
Switched to `*string`.